tests/zfs-tests/tests/functional/snapshot/snapshot_009_pos.ksh

```diff
@@ -52,18 +52,15 @@ function cleanup
 	typeset snap
 
 	for ds in $ctr/$TESTVOL1 $ctr/$TESTCLONE; do
-		datasetexists $ds && \
-			log_must zfs destroy -f $ds
+		destroy_dataset $ds "-rf"
```

Comment:
Preserve the `datasetexists $ds &&` part?

Reply:
We could keep it if you prefer, I choose to drop it because `destroy_dataset` includes the same check.

Reply:
Ah, no. This is fine - I misread the `log_note` in that function as a `log_fail`